crates/provider-sqldb-postgres/wit/deps/postgres/query.wit

```diff
@@ -0,0 +1,102 @@
+package wasmcloud:postgres@0.1.0-draft;
+
+/// Interface for querying a Postgres database
+interface query {
+  use types.{pg-value, result-row};
+  use connection.{connection-token};
+
+  /// Errors that occur while executing queries
+  variant query-error {
+    /// Unknown/invalid query parameters
+    invalid-params(string),
+    /// Invalid/malformed query
+    invalid-query(string),
+    /// A completely unexpected error, specific to executing queries
+    unexpected(string),
+  }
+
+  /// Query a Postgres database given a connection token (which can represent a connection *or* session)
+  ///
+  /// Queries *must* be parametrized, with named arguments in the form of `$<integer>`, for example:
+  ///
+  /// ```
+  /// SELECT email,username FROM users WHERE uuid=$1;
+  /// ```
+  ///
+  /// NOTE: To see how to obtain a `connection-token`, see `connection.wit`.
+  ///
+  query: func(token: connection-token, query: string, params: list<pg-value>) -> result<list<result-row>, query-error>;
+}
+
+/// Managed queries are similar to regular queries, except that clients trust the
+/// provider/implementer to completely manage all connection/session semantics and information.
+///
+/// Connections must be set up and managed out-of-band, completely unknown to the calling component.
+interface managed-query {
+  use types.{pg-value, result-row};
+  use query.{query-error};
```

Comment:
If we do keep the `managed-query` interface around, we should probably break out the `query-error` type into the types package. Otherwise if an implementor only wants to implement the `managed-query` interface, they'll still need to have a dependency on/implementation of the `query` interface

Reply:
Sure! moving all errors to `types`